### ISSUE TITLE
Question about support for libyang 3.x

### ISSUE THREAD

**Issue by irfanHaslanded** [sysrepo/sysrepo-cpp#23]
Hello,
I would like to know if/when the sysrepo-cpp and libyang-cpp packages be built using libyang 3.x

**Comment by jktjkt** [comment]
As far as I know the v3 API has not hit the upstream devel branch yet. Once that happens (and assuming that there are no "big breaking changes"), we will very likely follow suite.

There's one problem which is the rework of the auth backend in libnetconf2/netopeer2-server. We're relying on an auth method which uses either SSH keys, or kbd-interactive, or password authentication, allowing any of these to succeed. So far, it's been OK to run the devel branch of libyang + sysrepo with a slightly older Netopeer2. If that's not the case in libyang v3, we will have some PAM code backporting to do.

TL;DR: if you need this, please submit a patch so that others can benefit from the port. We might not merge it as-is right now, but it's good to have code ready and publicly available.

**Comment by irfanHaslanded** [comment]
> As far as I know the v3 API has not hit the upstream devel branch yet.

Do you mean `master` branch?

**Comment by jktjkt** [comment]
Ah, my bad -- I was looking for a merge commit, but they rebased. Patches welcome, we're currently busy with some internal deadlines.

**Comment by jktjkt** [comment]
just fyi, the patches are up for grabs/testing/feedback here: https://gerrit.cesnet.cz/q/topic:libyang-v3

**Comment by jktjkt** [comment]
Merged.
